test/create_zarr_optimus.py

```diff
@@ -0,0 +1,307 @@
+        cell_metric_values.append(row_values)
+
+    # Gene metric values, the row and column sizes
+    ncols = len(cell_metrics[0][1:])
+    nrows = len(cell_metrics[1:])
+
+    # now insert the dataset that has the numeric values for the cell qc metrics
+    data_group.create_dataset(
+        "qc_cell_values",
+        shape=(nrows, ncols),
+        compressor=COMPRESSOR,
+        dtype=np.float32,
+        chunks=(nrows, ncols),
+        data=cell_metric_values)
+
+
+def add_expression_counts(data_group, args):
+    """Converts  the count matrix from the Optimus pipeline to zarr file
+    Args:
+        data_group (zarr.hierarchy.Group): datagroup object for the zarr
+        args (argparse.Namespace): input arguments for the run
+    """
+    # read the cell ids and adds into the cell_barcodes dataset
+    barcodes = np.load(args.cell_ids)
+
+    if len(barcodes):
+        data_group.create_dataset(
+            "cell_id",
+            shape=(1, len(barcodes)),
+            compressor=COMPRESSOR,
+            dtype='<U40',
+            chunks=(1, len(barcodes)),
+            data=[list(barcodes)])
+
+    # read the gene ids  and adds into the gene_ids dataset
+    gene_ids = np.load(args.gene_ids)
+
+    if len(gene_ids):
+        data_group.create_dataset(
+            "gene_id",
+            shape=(1, len(gene_ids)),
+            compressor=COMPRESSOR,
+            dtype='<U40',
+            chunks=(1, len(gene_ids)),
+            data=[list(gene_ids)])
+
+    # read .npz file expression counts and add it to the expression_counts dataset
+    exp_counts = np.load(args.count_matrix)
+
+    # now convert it back to a csr_matrix object
+    csr_exp_counts = sparse.csr_matrix((exp_counts['data'],
+                                        exp_counts['indices'],
+                                        exp_counts['indptr']),
+                                       shape=exp_counts['shape'])
+
+    # now create a dataset of zeros with the same dimensions as the expression count matrix
+    exp_counts_group = data_group.zeros('expression',
+                                        compressor=COMPRESSOR,
+                                        shape=csr_exp_counts.shape,
+                                        chunks=(CHUNK_ROW_SIZE, csr_exp_counts.shape[1]),
+                                        dtype=np.float32)
+
+    # load a chunks from the expression count matrix and update the corresponding chunk in expression matrix in zar
+    for i in range(0, csr_exp_counts.shape[0], CHUNK_ROW_SIZE):
+        # check if it is possible to make a full chunk of data
+        if i + CHUNK_ROW_SIZE <= csr_exp_counts.shape[0]:
+            exp_counts_group[i:i + CHUNK_ROW_SIZE, ] = csr_exp_counts[i:i + CHUNK_ROW_SIZE, ].toarray()
+        else:
+            # not enough data remaining, so make the final set of data less than a chunk
+            j = csr_exp_counts.shape[0] - i
+            exp_counts_group[i:i + j, ] = csr_exp_counts[i:i + j, ].toarray()
```

Comment:
Since `exp_counts_group .shape` == `csr_exp_counts .shape`, I'm not sure if the `j` here is necessary or not, can we do:
```python
exp_counts_group[i:, ] = csr_exp_counts[i:, ].toarray()
```
or 
```python
exp_counts_group[i:csr_exp_counts.shape[0], ] = csr_exp_counts[i:csr_exp_counts.shape[0], ].toarray()
```
instead to save the computation time?